### PR TITLE
aspects: support content sub-rules

### PR DESCRIPTION
**Pull request by MiguelPires**
Adds support for "content" rules that can be used to extend the rules in which they're nested. This is intended to be syntactic sugar equivalent to help write nested rules without having to repeat the parent prefix. Using the content rule is equivalent to adding a new rule with the prefix of the parent's request and storage plus whatever is in the content rule's own request and storage. These rules can also have access modes that are different than the rules that contain them.

### CODE REVIEW

**Review by andrewphelpsj — COMMENTED** [review]
One thing about making the "request" field optional, otherwise looks good

File: aspects/aspects.go [hunk 1]
```diff
@@ -297,6 +256,75 @@ func newAspect(bundle *Bundle, name string, aspectRules []interface{}) (*Aspect,
 	return aspect, nil
 }
 
+func parseRule(parent *aspectRule, ruleRaw interface{}) ([]*aspectRule, error) {
+	ruleMap, ok := ruleRaw.(map[string]interface{})
+	if !ok {
+		return nil, errors.New("each aspect rule should be a map")
+	}
+
+	requestRaw, ok := ruleMap["request"]
```
[the remark this answers is not in the document]
Comment (MiguelPires) [reply aspects/aspects.go:265]: Yes, it should default to storage but hasn't been implemented in the "normal" rules either so I didn't enforce it here

**Review by andrewphelpsj — APPROVED** [review]
Thanks!

File: aspects/aspects.go [hunk 2]
```diff
@@ -37,26 +37,21 @@ import (
 type accessType int
 
 const (
-	readWrite accessType = iota
+	readWrite accessType = 1 << iota
```
Comment (pedronis) [inline aspects/aspects.go:40]: why this change?

**Review by pedronis — COMMENTED** [review]
question

File: aspects/aspects.go (hunk 2 above)
Comment (MiguelPires) [reply aspects/aspects.go:40]: I think it might've been to simplify the check from a switch to `rule.access&(read|readWrite) != 0`. But reading it now, it seems unnecessary

**Review by pedronis — APPROVED** [review]
thanks
